libvast/src/meta_index.cpp

```diff
@@ -177,13 +191,28 @@ std::vector<uuid> meta_index::lookup(const expression& expr) const {
         for (auto& [part_id, part_syn] : synopses_) {
           VAST_DEBUG(this, "checks", part_id, "for predicate", x);
           for (auto& [field, syn] : part_syn.field_synopses_) {
-            if (syn && match(field)) {
-              found_matching_synopsis = true;
-              auto opt = syn->lookup(x.op, make_view(rhs));
-              if (!opt || *opt) {
-                VAST_DEBUG(this, "selects", part_id, "at predicate", x);
-                result.push_back(part_id);
-                break;
+            if (match(field)) {
+              if (syn) {
+                found_matching_synopsis = true;
+                auto opt = syn->lookup(x.op, make_view(rhs));
+                if (!opt || *opt) {
+                  VAST_DEBUG(this, "selects", part_id, "at predicate", x);
+                  result.push_back(part_id);
+                  break;
+                }
+              } else {
+                // The field has no dedicated synopsis. Check if there is one
+                // for the type in general.
+                auto it = part_syn.type_synopses_.find(field.type);
```

Comment:
This assumes exact type equality. Two string types with different attributes are not equal wrt. `operator==`. You may want to strip attributes here.

Take a look a the `key` function in our `factory` implementations, where we do this as well.